lib/smile-identity-core/business_verification.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'base64'
+require 'openssl'
+require 'uri'
+require 'typhoeus'
+require_relative 'validations'
+
+module SmileIdentityCore
+  ##
+  # The business verification product lets you search the business registration or
+  # tax information (available in Nigeria only) of a business from one of our supported countries.
+  # For more info visit https://docs.smileidentity.com/products/for-businesses-kyb/business-verification
+  class BusinessVerification
+    include Validations
+
+    BASIC_BUSINESS_REGISTRATION = 'BASIC_BUSINESS_REGISTRATION'
+    BUSINESS_REGISTRATION = 'BUSINESS_REGISTRATION'
+    TAX_INFORMATION = 'TAX_INFORMATION'
+
+    REQUIRED_ID_INFO_FIELD = %i[country id_type id_number].freeze
```

Comment:
there are a few conditional fields here:

- `business_type` for `BASIC_BUSINESS_REGISTRATION` and `BUSINESS_REGISTRATION` in NG
- `postal_code` and `postal_address` for KE `BUSINESS_REGISTRATION`

seeing as we handle this validation on the REST API, should we be replicating validation here, and if yes, how much is required?

Reply:
we can to fail fast before hitting the api then. Not much work will be required

Reply:
business_type is currently the only conditional field listed in the docs, there's currently no entry for postal_code and postal_address

Reply:
that's going with KE shortly. 

this supports my point: the server to server libraries should be light wrappers around the API, that way new country / ID type support does not require an update to the Resource Definition

Reply:
I see, since the requirements can be added and removed based on business needs. We have to reduce the need to update the libraries based on these

We can do basic validations and defer the more granular ones to the api

Reply:
For now, country id_type id_number can be validated since they are the fundamental data we collect

Reply:
works for me